src/test/java/com/example/test/TestApplicationDataTests.java

```diff
@@ -0,0 +1,52 @@
+package com.example.test;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import com.example.test.customer.CustomerRepository;
+import com.example.test.project.ProjectRepository;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = Replace.NONE)
+@Testcontainers(disabledWithoutDocker = true)
+class TestApplicationDataTests {
+    @Container
+    public static MySQLContainer<?> mySQLContainer = new MySQLContainer<>("mysql:8.0.33")
+            .withFileSystemBind("./mysql_test_container_volume", "/var/lib/mysql", BindMode.READ_WRITE)
+            .withStartupAttempts(1)
+            .withStartupTimeoutSeconds(1200)
+            .withDatabaseName("my_app")
+            .withUsername("my_app");
```

Comment:
I have two points to make:

when I apply your changes, the test gets stuck at 'Waiting for database connection to become available at ...'. However, when I check out your fork, it runs through successfully as expected.

The problem was solved by removing the `.withDatabaseName()` and `.withUserName()` and adjusting the `application.yml` `username` to root.

```suggestion
    @Container
    public static MySQLContainer<?> mySQLContainer = new MySQLContainer<>("mysql:8.0.33")
            .withFileSystemBind("./mysql_test_container_volume", "/var/lib/mysql", BindMode.READ_WRITE)
            .withStartupAttempts(1)
            .withStartupTimeoutSeconds(1200);
```

Reply:
yes, the "my_app/my_app" (sorry, for that) is fixed to|conforms with application.yaml (keep complexity low..) AND (gracefully) combines with "(static) dynamic properties":)

Reply:
i think the "system bind" improves testing time, but it's a "double-edged sword" :) (keep you changleogs & test data (clean and) consistent...otherwise: `rm -rf ./mysql_test_container_volume` 😸